### PR TITLE
GFORMS-1302 ATL mini check your answers

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala [hunk 1]
```diff
@@ -121,6 +127,67 @@ class SectionRenderingService(frontendAppConfig: FrontendAppConfig, lookupRegist
...
+            sectionTitle4Ga,
+            cache.form.thirdPartyData.obligations,
+            validationResult,
+            envelope,
+            None,
+            FastForward.StopAt(sectionNumber)
+          )
+      }
+    }
+    val edit = request.getQueryString("edit").fold(false)(_.toBoolean)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala:172]: `toBoolean` is not safe. Let's wrap it in `Try` os something.

**Review by josef-vlach — CHANGES_REQUESTED** [review]
It is crashing when `defaultPage` is defined. It crashes when second cya should be displayed.

**Comment by girishkamat** [comment]
@josef-vlach I had to filter the invisible iterations from the render page model, to get it working (all changes in FormController). Does that look ok?

File: app/uk/gov/hmrc/gform/gform/FormController.scala [hunk 2]
```diff
@@ -151,32 +150,34 @@ class FormController(
...
                 iteration.checkYourAnswers match {
                   case Some(checkYourAnswers) if checkYourAnswers.sectionNumber == sectionNumber =>
-                    val visibleIteration = iteration.copy(singletons =
-                      NonEmptyList.fromListUnsafe(
-                        iteration.singletons.filter(s =>
-                          formModelOptics.formModelVisibilityOptics.formModel.availableSectionNumbers
-                            .contains(s.sectionNumber)
-                        )
+                    val maybeVisibleIteration: Option[Bracket.AddToListIteration[Visibility]] =
+                      formModelOptics.formModelVisibilityOptics.formModel.addToListBrackets.collectFirst {
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/FormController.scala:154]: Can't you use `def bracket(sectionNumber: SectionNumber): Bracket[A]` defined on `FormModel`?

File: app/uk/gov/hmrc/gform/gform/FormController.scala [hunk 3]
```diff
@@ -151,9 +151,11 @@ class FormController(
                 iteration.checkYourAnswers match {
                   case Some(checkYourAnswers) if checkYourAnswers.sectionNumber == sectionNumber =>
                     val maybeVisibleIteration: Option[Bracket.AddToListIteration[Visibility]] =
-                      formModelOptics.formModelVisibilityOptics.formModel.addToListBrackets.collectFirst {
-                        case b if b.hasSectionNumber(sectionNumber) => b.iterationForSectionNumber(sectionNumber)
-                      }
+                      formModelOptics.formModelVisibilityOptics.formModel
+                        .bracket(sectionNumber)
+                        .fold[Option[Bracket.AddToListIteration[Visibility]]](_ => None)(_ => None)(a =>
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/FormController.scala:156]: You can use `withAddToListBracket` method on `Bracket`

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**